### ISSUE TITLE
Keyboard input issue with German layout on Windows using AltGr combinations on Chrome

### ISSUE THREAD

**Issue by wintamute** [novnc/noVNC#1062]
I'm having issues with German keyboard layout, using latest novnc (commit 24231f) on Openstack, using latest Chrome. All keys using AltGr combinations show just the normal key.

Client OS: Win10 x64
Client Browser: Chrome 66 x64
VM OS: Windows Server 2016 x64

When using Chrome (Chromium 64.0.3282.167) on a Linux Client OS (Ubuntu 16.04 x64), everything is fine.

I also used this tool to check what the browsers actually do:[ https://w3c.github.io/uievents/tools/key-event-viewer.html]( https://w3c.github.io/uievents/tools/key-event-viewer.html)
I attached a screenshot of the output, test was holding down AltGr for about a second, then pressing 7,8,9,0, then releasing AltGr.

If I can help in any way please let me know.

![keys_chrome](https://user-images.githubusercontent.com/11716734/39130434-5a342a4a-470d-11e8-8045-0af67b00d67e.PNG)

**Comment by CendioOssman** [comment]
I don't have an Openstack setup to test with here, but with TigerVNC this works just fine.

Could you configure noVNC's log level to `info` and see what you get in the browser console when you do the same AltGr test?

**Comment by wintamute** [comment]
I played around a bit more which involved restarting the Openstack services/rebooting the host machine and VM and for some odd reason everything now works as expected, even when using latest Firefox.

Just for completeness, the setup is Openstack Queens release running on Ubuntu 16.04, using kvm with qemu with builtin vnc and latest noVNC from a git clone, replacing the the one from Ubuntu package (which is from around 2013).

Closing this issue.
